Windows warnings about inconsistent dll linkage when building runGui.exe

## Environment
* OS Version: Jenkins windows CI machines
* Source or binary build? source build of gz-sim7


## Description
* Expected behavior: build should compile without warnings
* Actual behavior: some windows compiler warnings
    * [![Build Status](https://build.osrfoundation.org/buildStatus/icon?job=gz_sim-7-win&build=3)](https://build.osrfoundation.org/view/ign-garden/job/gz_sim-7-win/3/) https://build.osrfoundation.org/view/ign-garden/job/gz_sim-7-win/3/msbuild/

## Steps to reproduce
<!-- Provide steps so we can try to reproduce this issue -->

1. Run https://build.osrfoundation.org/view/ign-garden/job/gz_sim-7-win
2.
3.

## Output

~~~
  gz.cc

C:\J\workspace\gz_sim-7-win\ws\gz-sim\src\gz.cc(46,1): warning C4273: 'gzSimVersion': inconsistent dll linkage [C:\J\workspace\gz_sim-7-win\ws\build\gz-sim7\src\runGui.vcxproj]

C:\J\workspace\gz_sim-7-win\ws\gz-sim\src\gz.hh(24,36): message : see previous definition of 'gzSimVersion' [C:\J\workspace\gz_sim-7-win\ws\build\gz-sim7\src\runGui.vcxproj]

C:\J\workspace\gz_sim-7-win\ws\gz-sim\src\gz.cc(52,1): warning C4273: 'simVersionHeader': inconsistent dll linkage [C:\J\workspace\gz_sim-7-win\ws\build\gz-sim7\src\runGui.vcxproj]

C:\J\workspace\gz_sim-7-win\ws\gz-sim\src\gz.hh(28,36): message : see previous definition of 'simVersionHeader' [C:\J\workspace\gz_sim-7-win\ws\build\gz-sim7\src\runGui.vcxproj]

C:\J\workspace\gz_sim-7-win\ws\gz-sim\src\gz.cc(59,1): warning C4273: 'cmdVerbosity': inconsistent dll linkage [C:\J\workspace\gz_sim-7-win\ws\build\gz-sim7\src\runGui.vcxproj]

C:\J\workspace\gz_sim-7-win\ws\gz-sim\src\gz.hh(32,35): message : see previous definition of 'cmdVerbosity' [C:\J\workspace\gz_sim-7-win\ws\build\gz-sim7\src\runGui.vcxproj]

C:\J\workspace\gz_sim-7-win\ws\gz-sim\src\gz.cc(73,1): warning C4273: 'worldInstallDir': inconsistent dll linkage [C:\J\workspace\gz_sim-7-win\ws\build\gz-sim7\src\runGui.vcxproj]

C:\J\workspace\gz_sim-7-win\ws\gz-sim\src\gz.hh(35,42): message : see previous definition of 'worldInstallDir' [C:\J\workspace\gz_sim-7-win\ws\build\gz-sim7\src\runGui.vcxproj]

C:\J\workspace\gz_sim-7-win\ws\gz-sim\src\gz.cc(81,1): warning C4273: 'findFuelResource': inconsistent dll linkage [C:\J\workspace\gz_sim-7-win\ws\build\gz-sim7\src\runGui.vcxproj]

C:\J\workspace\gz_sim-7-win\ws\gz-sim\src\gz.hh(87,42): message : see previous definition of 'findFuelResource' [C:\J\workspace\gz_sim-7-win\ws\build\gz-sim7\src\runGui.vcxproj]

C:\J\workspace\gz_sim-7-win\ws\gz-sim\src\gz.cc(143,1): warning C4273: 'runServer': inconsistent dll linkage [C:\J\workspace\gz_sim-7-win\ws\build\gz-sim7\src\runGui.vcxproj]

C:\J\workspace\gz_sim-7-win\ws\gz-sim\src\gz.hh(62,34): message : see previous definition of 'runServer' [C:\J\workspace\gz_sim-7-win\ws\build\gz-sim7\src\runGui.vcxproj]

C:\J\workspace\gz_sim-7-win\ws\gz-sim\src\gz.cc(424,1): warning C4273: 'runGui': inconsistent dll linkage [C:\J\workspace\gz_sim-7-win\ws\build\gz-sim7\src\runGui.vcxproj]

C:\J\workspace\gz_sim-7-win\ws\gz-sim\src\gz.hh(80,34): message : see previous definition of 'runGui' [C:\J\workspace\gz_sim-7-win\ws\build\gz-sim7\src\runGui.vcxproj]

  gz.obj : MSIL .netmodule or module compiled with /GL found; restarting link with /LTCG; add /LTCG to the link command line to improve linker performance

     Creating library C:/J/workspace/gz_sim-7-win/ws/build/gz-sim7/lib/Release/runGui.lib and object C:/J/workspace/gz_sim-7-win/ws/build/gz-sim7/lib/Release/runGui.exp

  Generating code

  Finished generating code

  runGui.vcxproj -> C:\J\workspace\gz_sim-7-win\ws\build\gz-sim7\bin\Release\runGui.exe
~~~

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -168,8 +168,8 @@ target_link_libraries(${gz_lib_target}
 )
 
 # Executable target that runs the GUI without ruby for debugging purposes.
-add_executable(runGui gz.cc)
-target_link_libraries(runGui PRIVATE ${gz_lib_target})
+add_executable(runGui cmd/runGui_main.cc)
+target_link_libraries(runGui PRIVATE gz ${gz_lib_target})
 
 # Create the library target
 gz_create_core_library(SOURCES ${sources} CXX_STANDARD 17)

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -169,7 +169,7 @@ target_link_libraries(${gz_lib_target}
 
 # Executable target that runs the GUI without ruby for debugging purposes.
 add_executable(runGui cmd/runGui_main.cc)
-target_link_libraries(runGui PRIVATE gz ${gz_lib_target})
+target_link_libraries(runGui PRIVATE ${gz_lib_target})
 
 # Create the library target
 gz_create_core_library(SOURCES ${sources} CXX_STANDARD 17)

--- a/src/cmd/runGui_main.cc
+++ b/src/cmd/runGui_main.cc
@@ -1,0 +1,6 @@
+#include "gz/sim/gui/Gui.hh"
+
+int main(int argc, char* argv[])
+{
+  return gz::sim::gui::runGui(argc, argv, nullptr);
+}

--- a/src/cmd/runGui_main.cc
+++ b/src/cmd/runGui_main.cc
@@ -1,3 +1,20 @@
+/*
+ * Copyright (C) 2023 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
 #include "gz/sim/gui/Gui.hh"
 
 int main(int argc, char* argv[])

--- a/src/gz.cc
+++ b/src/gz.cc
@@ -452,8 +452,3 @@ extern "C" int runGui(const char *_guiConfig, const char *_file, int _waitGui,
   return gz::sim::gui::runGui(
     argc, argv, _guiConfig, _file, _waitGui, _renderEngine);
 }
-
-int main(int argc, char* argv[])
-{
-  return sim::gui::runGui(argc, argv, nullptr);
-}

